### DISCUSSION TITLE
MachineDisks.device.partion Mounted. How K8s Local Storage It?

### DISCUSSION THREAD

**Discussion by mikeU-1F45F** [siderolabs/talos#6221] [Q&A] [answered]
Hey Talos Rockstars,

I'm really lov'n Talos from top to bottom.  Excellent work, All!  Thanks!

Situation:
I've set up a single-node cluster NUC with Talos running on /dev/sda and a secondary /dev/sdb mounted successfully at `/var/mnt/storage` using MachineDisks.  I can see it in `talosctl list /var/mnt` and `talosctl mounts` 👍

I'm trying to use K8s Local PersistentVolume to drop files onto it following [random guides](https://lapee79.github.io/en/article/use-a-local-disk-by-local-volume-static-provisioner-in-kubernetes/) on the Internet. When I `kubectl apply -f` a deployment yaml, the pod's `describe` points out 

`Warning  FailedMount  2m38s (x33 over 53m)  kubelet  MountVolume.NewMounter initialization failed for volume "pv-data" : path "/var/mnt/storage/" does not exist` 

Kubernetes is not "seeing" the Talos mount. How do I tie together this local storage from Talos so K8s can use it?

Thanks in advance and Cheers.

**Comment by smira** [comment, marked as answer]
You might need to bind-mount the path into the kubelet container on Talos, so that kubelet sees the path. 

Like in the [example](https://www.talos.dev/v1.2/reference/configuration/#extramount), but with your path

**Reply by mikeU-1F45F** [reply]
Yup. That did it.  I'll read up and tweak the extraMounts.options a bit, but I see the K8s deployment writing to that mount now.  Thanks @smira
